src/praisonai/praisonai/bots/_mirror.py

```
@@ -0,0 +1,134 @@
+                "mirror_source": source_label,
+            }
+            if metadata:
+                # Filter out reserved fields to preserve entry invariants
+                _reserved = {"role", "content", "timestamp", "mirror", "mirror_source"}
+                entry.update({k: v for k, v in metadata.items() if k not in _reserved})
+            history.append(entry)
+            
+            # Save updated history atomically within the lock
+            session_mgr._save_history(user_id, history)
+            
+        except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)